### PR TITLE
Fix: Map relations to snapshot physical tables in the dbt adapter

### PR DESCRIPTION
**Pull request by izeigerman**
Sometimes `ref` are passed into the dbt adapter while constructing a model query. This means that the production table name will be used since jinja rendering happens before we get a chance to resolve table references. 

This update ensures that dbt's `ref` returns resolved table references when using adapter.

### CODE REVIEW

File: sqlmesh/core/renderer.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ def __init__(
         jinja_macro_registry: t.Optional[JinjaMacroRegistry] = None,
         python_env: t.Optional[t.Dict[str, Executable]] = None,
         only_latest: bool = False,
+        resolve_tables: bool = True,
```
Comment (crericha) [inline sqlmesh/core/renderer.py:64]: Can we push resolution of tables to later all the time? I'm wondering if that will simplify the code (assuming its possible)

File: sqlmesh/dbt/builtin.py [hunk 2]
```diff
@@ -162,13 +171,33 @@ def var(name: str, default: t.Optional[t.Any] = None) -> t.Any:
...
+            logger.warning("Could not resolve ref '%s'", ref_name)
             return None
 
+        relation_info = relation_info.copy()
+        model_name = ".".join(
+            relation_info[p] for p in ("database", "schema", "identifier") if relation_info.get(p)
+        )
+        table_mapping = to_table_mapping(snapshots.values(), is_dev)
+        if model_name in table_mapping:
+            physical_table_name = table_mapping[model_name]
```
Comment (crericha) [inline sqlmesh/dbt/builtin.py:194]: Is it possible that the mapping isn't the dict? If so, should we catch and raise a better error message?

File: sqlmesh/dbt/builtin.py [hunk 3]
```diff
@@ -162,13 +171,33 @@ def var(name: str, default: t.Optional[t.Any] = None) -> t.Any:
...
+            logger.info("Resolved ref '%s' to snapshot table '%s'", ref_name, physical_table_name)
```
Comment (crericha) [inline sqlmesh/dbt/builtin.py:195]: Maybe debug instead of info?

File: sqlmesh/dbt/builtin.py [hunk 4]
```diff
@@ -178,6 +207,7 @@ def generate_source(sources: t.Dict[str, t.Any], api: Api) -> t.Callable:
     def source(package: str, name: str) -> t.Optional[BaseRelation]:
         relation_info = sources.get(f"{package}.{name}")
         if relation_info is None:
+            logger.warning("Could not resolve source '%s.%s'", package, name)
```
Comment (crericha) [inline sqlmesh/dbt/builtin.py:210]: Package isn't schema, so "package.name" doesn't really make sense. Recommend package=, name=.

File: sqlmesh/core/renderer.py (hunk 1 above)
Comment (izeigerman) [reply sqlmesh/core/renderer.py:64]: Not sure I understand what you're suggesting exactly

File: sqlmesh/dbt/builtin.py (hunk 2 above)
Comment (izeigerman) [reply sqlmesh/dbt/builtin.py:194]: When mapping can not be a dict?

Comment (crericha) [reply sqlmesh/dbt/builtin.py:194]: I meant the mapping isn't *in* the dict. Key word missing there!

File: sqlmesh/core/renderer.py (hunk 1 above)
Comment (crericha) [reply sqlmesh/core/renderer.py:64]: My understanding is this boolean denotes whether we should resolve tables here or not. I'm assuming that means we resolve them later? And if so, why wouldn't we always do it there.

Comment (izeigerman) [reply sqlmesh/core/renderer.py:64]: Actually, let me refactor

Comment (izeigerman) [reply sqlmesh/core/renderer.py]: refactored, please take a look

**Review by crericha — APPROVED** [review]
I like this approach much more. About as clean as we can get I think.

**Comment by izeigerman** [comment]
@crericha thanks for suggesting it in the first place!

**Pull request merged**